grails-app/controllers/com/unifina/controller/api/IntegrationKeyApiController.groovy

```diff
@@ -65,6 +73,25 @@ class IntegrationKeyApiController {
 		render ""
 	}
 
+	@StreamrApi
+	def update(String id) {
+		Map json = new JsonSlurper().parseText((String) request.JSON)
+		if (json.name == null || json.name.trim() == "") {
+			response.status = 400
+			return
+		}
+		String name = (String) json.name
+
+		try {
+			ethereumIntegrationKeyService.updateKey(apiUser(), id, name)
+		} catch (NotFoundException e) {
+			response.status = 404
```

Comment:
No need to catch because `NotFoundException` extends `ApiException` and hence is serialized into a JSON error response (and status code) automagically be ErrorController.

Reply:
Status codes don't seem to be set by error controller! Which is odd because ErrorControllerSpec seems solid. Perhaps render() sets the response code anyway?